tests/test_types.py

```diff
@@ -1675,25 +1675,6 @@ class Model(BaseModel):
         ]
 
 
-@pytest.mark.parametrize(
-    'kwargs,type_',
-    [
-        ({'pattern': '^foo$'}, int),
-        ({'gt': 0}, conlist(int, min_length=4)),
-        ({'gt': 0}, conset(int, min_length=4)),
-        ({'gt': 0}, confrozenset(int, min_length=4)),
-    ],
-)
-def test_invalid_schema_constraints(kwargs, type_):
-    match = (
-        r'(:?Invalid Schema:\n.*\n  Extra inputs are not permitted)|(:?The following constraints cannot be applied to)'
-    )
-    with pytest.raises((SchemaError, TypeError), match=match):
-
-        class Foo(BaseModel):
-            a: type_ = Field('foo', title='A title', description='A description', **kwargs)
-
-
```

Comment:
```suggestion
@pytest.mark.parametrize(
    'kwargs,type_',
    [
        ({'gt': 0}, conlist(int, min_length=4)),
        ({'gt': 0}, conset(int, min_length=4)),
        ({'gt': 0}, confrozenset(int, min_length=4)),
    ],
)
def test_invalid_schema_constraints(kwargs, type_):
    match = (
        r'(:?Invalid Schema:\n.*\n  Extra inputs are not permitted)|(:?The following constraints cannot be applied to)'
    )
    with pytest.raises((SchemaError, TypeError), match=match):

        class Foo(BaseModel):
            a: type_ = Field('foo', title='A title', description='A description', **kwargs)
```

Reply:
The reason why the pattern one doesn't give an error anymore and the others do is that those go down the `__prepare_pydantic_annotations__` path. One of the important bits of this PR is that we might be able to get rid of that in the future, but I guess it makes sense to leave the tests in for now.

Reply:
I'll add this but just to be clear it's more documenting current behavior than testing the desired / future behavior